pkg/chunk/aws/dynamodb_table_client.go

```diff
@@ -2,46 +2,37 @@ package aws
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
-	"github.com/aws/aws-sdk-go/service/applicationautoscaling"
-	"github.com/aws/aws-sdk-go/service/applicationautoscaling/applicationautoscalingiface"
 	"github.com/aws/aws-sdk-go/service/dynamodb"
 	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
 	"github.com/go-kit/kit/log/level"
-	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/time/rate"
 
 	"github.com/weaveworks/common/instrument"
 	"github.com/weaveworks/cortex/pkg/chunk"
 	"github.com/weaveworks/cortex/pkg/util"
 )
 
-const (
-	autoScalingPolicyNamePrefix = "DynamoScalingPolicy_cortex_"
-)
-
-var applicationAutoScalingRequestDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-	Namespace: "cortex",
-	Name:      "application_autoscaling_request_duration_seconds",
-	Help:      "Time spent doing ApplicationAutoScaling requests.",
-
-	// AWS latency seems to range from a few ms to a few sec. So use 8 buckets
-	// from 128us to 2s. TODO: Confirm that this is the case for ApplicationAutoScaling.
-	Buckets: prometheus.ExponentialBuckets(0.000128, 4, 8),
-}, []string{"operation", "status_code"})
+// Pluggable auto-scaler implementation
+type autoscale interface {
+	CreateTable(ctx context.Context, desc chunk.TableDesc) error
```

Comment:
This is really `PostCreateTable`, right?

Reply:
yeah, I was wrestling with how similar it is to the higher-level interface